docs/source/integrations/triton.rst

```diff
@@ -1,3 +1,536 @@
+
+          .. tab-item:: CPU
+              :sync: cpu
+
+              .. code-block:: bash
+
+                  $ docker run --init --rm --network=host triton-integration:cpu serve-grpc \
+                                        --production --triton-options model-control-mode=explicit \
+                                        --triton-options load-model=onnx_mnist --triton-options load-model=torchscript_yolov5s
+
+.. tip::
+
+   To see all available options for Triton run:
+
+   .. code-block:: bash
+
+      $ docker run --init --rm -p 3000:3000 triton-integration:gpu tritonserver --help
+
+Current caveats
+~~~~~~~~~~~~~~~
+
+Versioning policy limitations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, model configuration `version policy <https://docs.nvidia.com/deeplearning/triton-inference-server/user-guide/docs/user_guide/model_configuration.html#version-policy>`_
+is set to ``latest(n=1)``, meaning the latest version of the model will be loaded into Triton server.
+
+Currently, TritonRunner only supports the ``latest`` policy.
+If you have multiple versions of the same model in your BentoService, then the runner only consider the latest version.
+
+For example, if the model repository have the following structure:
+
+.. code-block:: bash
+
+    model_repository
+    ├── onnx_mnist
+    │   ├── 1
+    │   │   └── model.onnx
+    │   ├── 2
+    │   │   └── model.onnx
+    │   └── config.pbtxt
+    ...
+
+.. epigraph::
+   Then ``triton_runner.onnx_mnist`` will reference to the latest version of the model (in this case, version 2).
+
+To use a specific version of said model, refer to the example below:
+
+.. literalinclude:: ./snippets/triton/service_model_version_1.py
+    :language: python
+    :caption: `service.py`
+
+Inference Protocol and Metrics Server
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, TritonRunner uses gRPC protocol that is defined in Inference protocol [#triton_inference_protocol]_.
+
+HTTP/REST APIs is disabled by default, though it can be enabled via ``BENTOML_USE_HTTP_TRITONSERVER`` environment variable.
+
+.. code-block:: bash
+
+    $ docker run --init --rm -p 3000:3000 triton-integration:cpu serve \
+                            --production --triton-options model-control-mode=explicit \
+                            --triton-options load-model=onnx_mnist --triton-options load-model=torchscript_yolov5s
+
+.. epigraph::
+    Currently, TritonRunner does not support running `Metrics server <https://docs.nvidia.com/deeplearning/triton-inference-server/user-guide/docs/user_guide/metrics.html>`_.
+    If you are interested in supporting the metrics server, please open an issue on :github:`GitHub <bentoml/BentoML/issues/new/choose>`
+
+Additionally, BentoML supervisors will allocate a random port for the gRPC server, hence ``grpc-port`` options that is passed to ``--triton-options`` will be omited.
+
+---------------
+
+Remarks
```

Comment:
Does this section and the section below belong in the documentation or engineering blog?

Reply:
I think we can move this to the blog.